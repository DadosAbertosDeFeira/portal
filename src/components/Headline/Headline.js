/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './Headline.module.scss';

const Headline = ({ className, size, children, testId, ...props }) => {
  switch (size) {
    case 'large': {
      return (
        <h1
          data-testid={testId}
          className={classnames(styles.headline, styles.large, className)}
          {...props}
        >
          {children}
        </h1>
      );
    }

    case 'small': {
      return (
        <h3
          data-testid={testId}
          className={classnames(styles.headline, styles.small, className)}
          {...props}
        >
          {children}
        </h3>
      );
    }

    default: {
      return (
        <h2
          data-testid={testId}
          className={classnames(styles.headline, styles.medium, className)}
          {...props}
        >
          {children}
        </h2>
      );
    }
  }
};

Headline.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf(['large', 'medium', 'small']),
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.array,
    PropTypes.object,
  ]),
};

Headline.defaultProps = {
  className: '',
  size: 'medium',
  children: '',
};

export default Headline;
