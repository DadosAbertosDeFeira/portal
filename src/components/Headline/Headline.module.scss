@import '../../styles/variables.scss';

.headline {
  color: $color-black;
  font-weight: 700;
  font-family: 'Montserrat', Roboto, sans-serif;
}

.medium {
  font-size: 1.5rem;
  line-height: 1.82rem;
}

.large {
  font-size: 3rem;
  line-height: 4.5rem;
}

.small {
  font-size: 1rem;
  line-height: 1.5rem;
}
