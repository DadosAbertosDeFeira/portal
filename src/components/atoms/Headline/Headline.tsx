import classNames from "classnames";
import type { ComponentPropsWithRef, ReactNode } from "react";
import { createElement, forwardRef } from "react";

export type HeadlineProps = ComponentPropsWithRef<"h1"> & {
  as?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
  children?: ReactNode;
};

export const Headline = forwardRef<HTMLHeadingElement, HeadlineProps>(
  function Headline({ as = "h1", children, className, ...props }, ref) {
    const classes = classNames(
      className,
      as === "h1" && "tablet:text-xl",
      "text-lg font-bold"
    );

    return createElement(as, { ...props, className: classes, ref }, children);
  }
);
