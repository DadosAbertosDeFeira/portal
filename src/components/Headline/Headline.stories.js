/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import Headline from './Headline';

export const HeadlineTemplate = (args) => <Headline {...args} />;
HeadlineTemplate.args = {
  children: 'Portal Dados Abertos',
  size: 'medium',
};
HeadlineTemplate.storyName = 'Default';

export const Small = HeadlineTemplate.bind({});
Small.args = {
  ...HeadlineTemplate.args,
  size: 'small',
};

export const Medium = HeadlineTemplate.bind({});
Medium.args = {
  ...HeadlineTemplate.args,
  size: 'medium',
};

export const Large = HeadlineTemplate.bind({});
Large.args = {
  ...HeadlineTemplate.args,
  size: 'large',
};

export default {
  title: 'Components/Headline',
  component: Headline,
};
