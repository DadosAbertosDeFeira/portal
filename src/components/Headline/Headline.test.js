/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { render } from '@testing-library/react';

import Headline from './Headline';

describe('<Headline />', () => {
  let sharedProps;

  beforeEach(() => {
    sharedProps = {
      testId: 'headline',
    };
  });

  it('renders correctly by default', () => {
    const { asFragment } = render(
      <Headline {...sharedProps}>Testing</Headline>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly large size', () => {
    const { asFragment } = render(
      <Headline {...sharedProps} size="large">
        Testing
      </Headline>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly small size', () => {
    const { asFragment } = render(
      <Headline {...sharedProps} size="small">
        Testing
      </Headline>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly medium size', () => {
    const { asFragment } = render(
      <Headline {...sharedProps} size="medium">
        Testing
      </Headline>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
