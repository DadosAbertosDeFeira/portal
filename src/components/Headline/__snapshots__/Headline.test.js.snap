// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> renders correctly by default 1`] = `
<DocumentFragment>
  <h2
    class="headline medium"
    data-testid="headline"
  >
    Testing
  </h2>
</DocumentFragment>
`;

exports[`<Headline /> renders correctly large size 1`] = `
<DocumentFragment>
  <h1
    class="headline large"
    data-testid="headline"
  >
    Testing
  </h1>
</DocumentFragment>
`;

exports[`<Headline /> renders correctly medium size 1`] = `
<DocumentFragment>
  <h2
    class="headline medium"
    data-testid="headline"
  >
    Testing
  </h2>
</DocumentFragment>
`;

exports[`<Headline /> renders correctly small size 1`] = `
<DocumentFragment>
  <h3
    class="headline small"
    data-testid="headline"
  >
    Testing
  </h3>
</DocumentFragment>
`;

exports[`<Headline /> renders correctly with html alternative attributes 1`] = `
<DocumentFragment>
  <h2
    class="headline medium"
    data-testid="headline"
    tabindex="0"
  >
    Testing
  </h2>
</DocumentFragment>
`;
